codegen/template_bundle/template_files.go

```diff
@@ -2197,6 +2199,23 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 	{{end -}}
 	sc.Peers().Add(ip + ":" + strconv.Itoa(int(port)))
 
+	var scAltName string
+	if deps.Default.Config.ContainsKey("clients.{{$clientID}}.staging.serviceName") {
+		scAltName = deps.Default.Config.MustGetString("clients.{{$clientID}}.staging.serviceName")
+		ipAlt := deps.Default.Config.MustGetString("clients.{{$clientID}}.staging.ip")
+		portAlt := deps.Default.Config.MustGetInt("clients.{{$clientID}}.staging.port")
+
+		scAlt := deps.Default.Channel.GetSubChannel(scAltName, tchannel.Isolated)
+		scAlt.Peers().Add(ipAlt + ":" + strconv.Itoa(int(portAlt)))
+	} else if deps.Default.Config.ContainsKey("clients.all.staging.serviceName") {
```

Comment:
The chance of a client is named `all` is rare, but it's possible. To avoid collision, we can use `clients.staging.all.serviceName` for the "all" case.

Reply:
Okay